dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/DefaultTrackerObjectsDeletionService.java

```diff
@@ -60,43 +66,49 @@ public class DefaultTrackerObjectsDeletionService implements TrackerObjectDeleti
 
   private final RelationshipService relationshipService;
 
-  private final EnrollmentTrackerConverterService enrollmentTrackerConverterService;
+  private final TrackedEntityAttributeValueService attributeValueService;
 
-  private final EventTrackerConverterService eventTrackerConverterService;
+  private final TrackedEntityDataValueChangeLogService dataValueChangeLogService;
+
+  private final ProgramNotificationInstanceService programNotificationInstanceService;
 
   @Override
-  public TrackerTypeReport deleteEnrollments(TrackerBundle bundle) throws NotFoundException {
+  public TrackerTypeReport deleteEnrollments(List<String> enrollments) throws NotFoundException {
+    UserInfoSnapshot userInfoSnapshot =
+        UserInfoSnapshot.from(CurrentUserUtil.getCurrentUserDetails());
     TrackerTypeReport typeReport = new TrackerTypeReport(TrackerType.ENROLLMENT);
 
-    List<org.hisp.dhis.tracker.imports.domain.Enrollment> enrollments = bundle.getEnrollments();
-
-    for (org.hisp.dhis.tracker.imports.domain.Enrollment value : enrollments) {
-      String uid = value.getEnrollment();
-
+    for (String uid : enrollments) {
       Entity objectReport = new Entity(TrackerType.ENROLLMENT, uid);
 
       Enrollment enrollment = manager.get(Enrollment.class, uid);
       if (enrollment == null) {
         throw new NotFoundException(Enrollment.class, uid);
       }
-      enrollment.setLastUpdatedByUserInfo(bundle.getUserInfo());
+      enrollment.setLastUpdatedByUserInfo(userInfoSnapshot);
+
+      List<String> events =
+          enrollment.getEvents().stream()
+              .filter(event -> !event.isDeleted())
```

Comment:
Are these `enrollment.getEvents()` coming from the payload? If so are we safe from users setting `deleted` on an event in the payload?

Reply:
These events are from the DB, the delete methods are only getting the ids of the entities to delete and then we retrieving them from the DB.